### ISSUE TITLE
[Feature request] Local-only / non-federated posts

### ISSUE THREAD

**Issue by SpencerDub** [friendica/friendica#7363]
Certain forks of the Mastodon project, such as [Hometown](https://github.com/hometown-fork/hometown) and [Mastodon Glitch Edition](https://glitch-soc.github.io/docs/) support a local-only scope for posts. Posts with the local-only scope do not propagate to other instances. You can read more discussion of this in [this Mastodon-related PR](https://github.com/tootsuite/mastodon/pull/8427).

Darius Kazemi of Hometown describes the benefits here:

> I've been running Friend Camp, a Mastodon fork with local only posting, for about a year. Being able to have conversations with people on your server that don't federate is a hugely liberating thing. It allows inside jokes to develop. It allows people the freedom to complain about things that they wouldn't necessarily feel comfortable leaving a trusted server (cops, employers, etc). It also lets us do things like have a server-wide movie night where we flood the local timeline with posts about the movie, and it doesn't pollute the rest of the Fediverse.

I recognize this can be sorta hacked together via Groups on Friendica, but that seems like vaguely convergent evolution. It's also a tough user experience: it would be far easier for me to tell my users, "Click the switch to keep your posts on this server" than "Make a list of all the people you are okay with the post going to, and be warned that if any of them are on another node, your data will leave the server". Groups have an important function, but I don't think they're best suited for this.

Does Friendica have a local-only/non-federated post scope option? Is there an add-on that adds this functionality? If not, please consider this a feature request for just that.

*(I'm relatively new to Friendica, am not a developer, and am in the pre-planning stages of setting up my own server, so I don't have a lot of technical knowledge that could help with this. That's why I'm linking to other projects of people who know more than me.)*

**Comment by MrPetovan** [comment]
Hi @SpencerDub and thank you for your interest in Friendica. There is a node-wide option to prevent any federation, but nothing on the post level as of yet.

I'm currently exploring meta-groups during my work on https://github.com/friendica/friendica/issues/7309. So far I've created the "Followers" and "Mutuals" meta-groups. They appear as groups in the post permission settings but they aren't actual physical groups any user would have to create. I suppose we can have a third "Local Followers" group that would comprise followers on the same node.

**Comment by SpencerDub** [comment]
Thanks for the welcome and the feedback, @MrPetovan! That "Local Followers" metagroup sounds like a great way to achieve this goal.

**Comment by SpencerDub** [comment]
I just installed 2019.12 successfully on my server and am starting to poke around! I'm excited to get my instance off the ground.

@MrPetovan, "Local Followers" is not yet supported as a meta-group, correct? I didn't see that option when I was drafting a post.

**Comment by tobiasd** [comment]
Yes, the "local followers" meta-group does currently not exist. But you can easily define your own groups of contacts for non-public postings.

**Comment by MrPetovan** [comment]
Re-reading the original post, it seems like there's also an avenue for adding a third option to the ACL selector. We currently have public and private with custom ACL, but we could add a middle-ground option which would be "local-only". It would be slightly more open than "Local Followers" in that anyone else on the same node could have access to it even without being a follower.

So these posts should be displayed on the local community page, but it should be clear if visitors can have access to them as well, even if remote servers/crawlers can't fetch them.

**Comment by SpencerDub** [comment]
I'm not much good for development, but is there any non-dev work I could do to help with this project?

**Comment by MrPetovan** [comment]
I understand, but both features are very different in terms of required work. "Local-only" is a very involved feature that doesn't exist yet, while "Local followers" is just a convenience shortcut in the ACL that can be emulated by a user group with handpicked contacts.

This is why I chose to focus on the "Local-only" feature. Not because there would be no value in the other, but rather because it can be more easily achieved independently, and probably warrants a separate issue that can be tackled way sooner than "Local-only".

**Comment by annando** [comment]
Possibly we could realize the "server public" with a new value for the `private` field. But I won't have time for the in the foreseeable future. My focus is currently at the database structure,

**Comment by SpencerDub** [comment]
Okay! I understand the project has multiple priorities and this isn't high on the list, especially since I'm the only one really pushing for it right now. :)

Thanks for the update, both of you, and @MrPetovan, thanks for explaining your reasoning.
